qir/qat/Rules/Factory.cpp

```diff
@@ -72,6 +72,8 @@ void RuleFactory::usingConfiguration(FactoryConfiguration const& config)
         optimizeResultZero();
     }
 
+    optimizeResultComparison();
```

Comment:
I think this fine for now, but we may want to match the style of the other parts where optimizing this comparison is gated by a config flag.

Reply:
Actually, that is not fine - this is a result of VSCode not saving because the file had changed on the harddrive - it should have had config around it.